### PR TITLE
[ENG-5173] Root folder picker

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-5173]
-   Feature flag: n/a

## Purpose
- Create root-folder-picker for configured storage addons

## Summary of Changes
- Make updates to mirage to support root-folder-picker

## Screenshot(s)
image

## QA Notes

[ENG-5173]: https://openscience.atlassian.net/browse/ENG-5173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/osf-components/addon/components/addons-service/manager/component.ts [hunk 1]
```diff
@@ -239,7 +236,8 @@ export default class AddonsServiceManagerComponent extends Component<Args> {
     @waitFor
     async confirmAccountSetup(account: AllAuthorizedAccountTypes) {
         if (this.selectedProvider && this.selectedAccount) {
-            await taskFor(this.selectedProvider.createConfiguredAddon).perform(account);
+            const newConfiguredAccount = await taskFor(this.selectedProvider.createConfiguredAddon).perform(account);
+            this.selectedConfiguration = newConfiguredAccount;
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/addons-service/manager/component.ts:240]: Why not just 
```suggestion
             this.selectedConfiguration = await taskFor(this.selectedProvider.createConfiguredAddon).perform(account);
```

File: lib/osf-components/addon/components/addons-service/root-folder-picker/component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @tracked selectedFolder = this.args.configuredStorageAddon.rootFolder;
+    @tracked currentItems: Item[] = [];
+
+    defaultKwargs = {
+        itemType: ItemType.Folder,
+    };
+
+    get disableSave() {
+        return this.selectedFolder === this.args.configuredStorageAddon.rootFolder;
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/addons-service/root-folder-picker/component.ts:29]: You could also disable save if save is running.

File: lib/osf-components/addon/components/addons-service/root-folder-picker/template.hbs [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<AddonsService::FileManager
+    @configuredStorageAddon={{@configuredStorageAddon}}
+    @defaultKwargs={{this.defaultKwargs}}
+    @startingFolderId={{this.selectedFolder}}
+    as |fileManager|
+>
+    <span>
+        <Button
+            @layout='fake-link'
+            {{on 'click' fileManager.goToRoot}}
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/addons-service/root-folder-picker/template.hbs:10]: Analytics and data-test selectors, please.

File: lib/osf-components/addon/components/addons-service/root-folder-picker/template.hbs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            <FaIcon @icon='folder' />
+            {{@configuredStorageAddon.rootFolder}}
+        </Button>
+    </span>
+    {{#each fileManager.currentPath as |pathItem|}}
+        <span>
+            <Button
+                @layout='fake-link'
+                aria-label={{t 'addons.configure.go-to-folder' folderName=pathItem.itemName}}
+                {{on 'click' (fn fileManager.goToFolder pathItem)}}
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/addons-service/root-folder-picker/template.hbs:21]: Here as well. And the other interactive controls.

File: mirage/scenarios/dashboard.ts [hunk 5]
```diff
@@ -101,39 +101,38 @@ export function dashboardScenario(server: Server, currentUser: ModelInstance<Use
 
     const boxAccount = server.create('authorized-storage-account', {
         displayName: 'My Box Account',
-        scopes: ['write'], // TODO: This should be a from an enum?
+        // scopes: ['write'], // TODO: This should be a from an enum?
```
Comment (brianjgeiger) [inline mirage/scenarios/dashboard.ts:104]: What's happening here?

Comment (futa-ikeda) [reply mirage/scenarios/dashboard.ts:104]: It looks like I commented these out to address later (`scopes` should have been renamed `authorizedCapabilities`), and forgot to address them. Fixing these now

File: tests/integration/components/addons-service/root-rolder-picker/component-test.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { setupIntl } from 'ember-intl/test-support';
+import { setupRenderingTest } from 'ember-qunit';
+import { TestContext } from 'ember-test-helpers';
+import { module, test } from 'qunit';
+import sinon, { SinonSpy } from 'sinon';
+
+import ExternalStorageServiceModel from 'ember-osf-web/models/external-storage-service';
+import ConfiguredStorageAddonModel from 'ember-osf-web/models/configured-storage-addon';
+// import root from 'ember-osf-web/mirage/factories/root';
```
Comment (brianjgeiger) [inline tests/integration/components/addons-service/root-rolder-picker/component-test.ts:13]: Oh no! Stray comment!

**Review by brianjgeiger — APPROVED** [review]
Aside from the stray comment, looks good. Yay tests!

**Pull request merged**